core/src/main/java/org/infinispan/configuration/parsing/ParseUtils.java

```diff
@@ -5,356 +5,421 @@
-        return unexpectedElement(reader, element.toString());
-    }
-
-    public static ConfigurationReaderException unexpectedElement(final ConfigurationReader reader, String element) {
-        return new ConfigurationReaderException("Unexpected element '" + element + "' encountered", reader.getLocation());
-    }
-
-    /**
-     * Get an exception reporting an unexpected end tag for an XML element.
-     * @param reader the stream reader
-     * @return the exception
-     */
-    public static ConfigurationReaderException unexpectedEndElement(final ConfigurationReader reader) {
-        return new ConfigurationReaderException("Unexpected end of element '" + reader.getLocalName() + "' encountered", reader.getLocation());
-    }
-
-    /**
-     * Get an exception reporting an unexpected XML attribute.
-     * @param reader the stream reader
-     * @param index the attribute index
-     * @return the exception
-     */
-    public static ConfigurationReaderException unexpectedAttribute(final ConfigurationReader reader, final int index) {
-        return new ConfigurationReaderException("Unexpected attribute '" + reader.getAttributeName(index) + "' encountered",
-                reader.getLocation());
-    }
-
-    /**
-     * Get an exception reporting an unexpected XML attribute.
-     * @param reader the stream reader
-     * @param name the attribute name
-     * @return the exception
-     */
-    public static ConfigurationReaderException unexpectedAttribute(final ConfigurationReader reader, final String name) {
-        return new ConfigurationReaderException("Unexpected attribute '" + name + "' encountered",
-                reader.getLocation());
-    }
-
-    /**
-     * Get an exception reporting an invalid XML attribute value.
-     * @param reader the stream reader
-     * @param index the attribute index
-     * @return the exception
-     */
-    public static ConfigurationReaderException invalidAttributeValue(final ConfigurationReader reader, final int index) {
-        return new ConfigurationReaderException("Invalid value '" + reader.getAttributeValue(index) + "' for attribute '"
-                + reader.getAttributeName(index) + "'", reader.getLocation());
-    }
-
-    /**
-     * Get an exception reporting a missing, required XML attribute.
-     * @param reader the stream reader
-     * @param required a set of enums whose toString method returns the
-     *        attribute name
-     * @return the exception
-     */
-    public static ConfigurationReaderException missingRequired(final ConfigurationReader reader, final Set<?> required) {
-        final StringBuilder b = new StringBuilder();
-        Iterator<?> iterator = required.iterator();
-        while (iterator.hasNext()) {
-            final Object o = iterator.next();
-            b.append(o.toString());
-            if (iterator.hasNext()) {
-                b.append(", ");
+   private ParseUtils() {
```

Comment:
Can you revert the format changes here? It makes it difficult to review and effectively means we lose the git history.

Reply:
Done. Unfortunately this file use 4-space indents...